src/card.rs

```diff
@@ -51,6 +62,19 @@ impl<T: Client> Card<T> {
         command: &iso7816::Command<C>,
         reply: &mut heapless::Vec<u8, R>,
     ) -> Result<(), Status> {
+        if let Some(reset_signal) = self.state.volatile.reset_signal {
+            match reset_signal.load() {
+                ResetSignal::None => {}
+                ResetSignal::ConfigChanged => {
+                    return Err(Status::SelectedFileInTerminationState);
+                }
+                ResetSignal::FactoryReset => {
+                    self.state = State::default();
```

Comment:
AFAIS this means we lose the reset signal.  Is this intended?

Reply:
Good catch.
The volatile state is also lost on reset. I'll put the signal in the Options instead.